test/test_nn_ops.cpp

```diff
@@ -300,51 +300,436 @@ TEST(NNOps, Conv2DBackpropFilterNHWCValid) {
+  vector<Tensor> ngraph_outputs;
+  opexecuter_ngraph.ExecuteOnNGraph(ngraph_outputs);
+
+  // Define scope for tf (without nGraph)
+  // Data Format: NHWC
+  Scope tf_scope = Scope::NewRootScope();
+  auto input_data_NHWC = ops::Const(tf_scope, input_size_NHWC);
+  auto output_delta_NHWC = ops::Transpose(tf_scope, output_delta, {0, 2, 3, 1});
+
+  auto r_tf =
+      ops::Conv2DBackpropInput(tf_scope, input_data_NHWC, filter,
+                               output_delta_NHWC, stride_NHWC, padding_type);
+
+  // Need to transpose the TF output to NCHW
+  auto tf_output_transposed = ops::Transpose(tf_scope, r_tf, {0, 3, 1, 2});
+  vector<Output> sess_run_fetchoutputs_tf = {tf_output_transposed};
+  OpExecuter opexecuter_tf(tf_scope, "Conv2DBackpropInput",
+                           static_input_indexes, output_datatypes,
+                           sess_run_fetchoutputs_tf);
+
+  vector<Tensor> tf_outputs;
+  opexecuter_tf.ExecuteOnTF(tf_outputs);
+
+  // Compare NGraph and TF Outputs
+  Compare(tf_outputs, ngraph_outputs);
+}  // end of op Conv2DBackpropInputNCHWSame
+
+// Conv2DBackpropInput op : compute the graidents of conv with respects to input
+// Input is in NCHW format, padding type = "SAME, with non-trivial dilation
+// attributes
+TEST(NNOps, Conv2DBackpropInputNCHWSameWithDilation) {
+  // Input NHWC :[batch, in_channels, in_height, in_width]
+  string padding_type = "SAME";
+  initializer_list<int32> input_size_NCHW = {1, 2, 7, 6};  // an integer vector
+
+  // Filter :[filter_height, filter_width, in_channels, out_channels]
+  vector<int64> filter_size_HWIO = {3, 3, 2, 2};
+  // Out_delta :[batch, out_channels, out_height, out_width]
+  vector<int64> output_del_size_same_NCHW = {1, 2, 4, 3};
+  std::vector<int> stride_NCHW = {1, 1, 2, 2};
+
+  // Conv2DBackpropInput has static input of index 0
+  vector<int> static_input_indexes = {0};
+
+  Scope ngraph_scope = Scope::NewRootScope();
+  ops::Conv2DBackpropInput::Attrs op_attr_nchw;
+  op_attr_nchw = op_attr_nchw.DataFormat("NCHW");
+  op_attr_nchw = op_attr_nchw.Dilations({1, 1, 3, 2});
+
+  auto input_data_NCHW = ops::Const(ngraph_scope, input_size_NCHW);
+
+  Tensor output_delta(DT_FLOAT, TensorShape(output_del_size_same_NCHW));
+  int size = 1;
+  for (auto num : output_del_size_same_NCHW) {
+    size *= num;
+  }
+  std::list<float> output_list(size);
+  std::iota(output_list.begin(), output_list.end(), 0);
+  std::vector<float> output_vector{std::begin(output_list),
+                                   std::end(output_list)};
+  AssignInputValues<float>(output_delta, output_vector);
+
+  Tensor filter(DT_FLOAT, TensorShape(filter_size_HWIO));
+  size = 1;
+  for (auto num : filter_size_HWIO) {
+    size *= num;
+  }
+  std::list<float> filter_list(size);
+  std::iota(filter_list.begin(), filter_list.end(), 0);
+  std::vector<float> filter_vector{std::begin(filter_list),
+                                   std::end(filter_list)};
```

Comment:
so `filter_vector` is just a copy of `filter_list` right? Why have 2 variables?


more minor comments:
```
int size_vect = std::accumulate(filter_size_HWIO .begin(), filter_size_HWIO .end(), 1, std::multiplies<float>());
std::vector<float> filter_vector();
std::iota(filter_vector.begin(), filter_vector.end(), 0);
```
the above snippet should take care of lines 404 to 411

similarly for lines 393 to 400

Reply:
Yes, `filter_vector` and `filter_list` are the same, was trying to convert a list to a vector, and then pass into the `AssignInputValues()` function.

Thanks for the code snippet, this is much simpler, I will rewrite all the similar parts using the `accumulate function`.

Reply:
Hey @sayantan-nervana , I eliminate the redundant `filter_list`, so there is only one variables instead of two. And I changed the `std::multiples<float>` to `std::multiple<int64>` in my latest change.